python/kserve/kserve/utils/utils.py

```diff
@@ -138,35 +138,18 @@ def to_headers(context: ServicerContext) -> Dict[str, str]:
     return headers
 
 
-def get_predict_input(payload: Union[Dict, InferRequest]) -> Union[np.ndarray, pd.DataFrame]:
+def get_predict_input(payload: Union[Dict, InferRequest], columns: List = None) -> Union[np.ndarray, pd.DataFrame]:
     if isinstance(payload, Dict):
         instances = payload["inputs"] if "inputs" in payload else payload["instances"]
         if len(instances) == 0:
             return np.array(instances)
-        if isinstance(instances[0], Dict):
+        if isinstance(instances[0], Dict) or (
+                isinstance(instances[0], List) and len(instances[0]) != 0 and isinstance(instances[0][0], Dict)):
             dfs = []
             for input in instances:
-                dfs.append(pd.DataFrame(input))
+                dfs.append(pd.DataFrame(input, columns=columns))
```

Comment:
@sivanantha321 so this works for both input formats with dict and list ?

Reply:
Yes

Reply:
Is columns default to None in pandas framework ?

Reply:
Yes